pkg/virtctl/imageupload/imageupload.go

```diff
@@ -172,6 +172,9 @@ func (c *command) run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	if !strings.HasPrefix(uploadProxyURL, "http") {
+		uploadProxyURL = fmt.Printf("http://$%s", uploadProxyURL)
```

Comment:
Maybe dumb question, but wouldn't we want to sue `https` as default nowadays?

Reply:
ok, then let's go with `http`. Can you check for `http://`?

Reply:
Maybe you can do something like `url.Parse(str)` and then checke if the `Scheme` is set.

Reply:
done

Reply:
Almost there. You will have to check if we see an error. For instance `htt@p://test` would fail and then we would try to access `u` which is `nil`.

Reply:
right, addressed